packages/medusa/src/types/product.ts

```diff
@@ -7,29 +7,26 @@ import {
   IsString,
   ValidateNested,
 } from "class-validator"
+import { ProductOptionValue, Status } from "../models"
 import { optionalBooleanMapper } from "../utils/validators/is-boolean"
 import { IsType } from "../utils/validators/is-type"
 import { DateComparisonOperator, StringComparisonOperator } from "./common"
 
-export enum ProductStatus {
-  DRAFT = "draft",
-  PROPOSED = "proposed",
-  PUBLISHED = "published",
-  REJECTED = "rejected",
-}
-
+/**
+ * API Level DTOs + Validation rules
+ */
 export class FilterableProductProps {
   @IsString()
   @IsOptional()
-  id?: string
+  id?: string | string[]
```

Comment:
question: Why this change? also, if it is an array it will not pass the constraint `@IsString`

Reply:
Because of this: https://github.com/medusajs/medusa/blob/f1b06958c2108ce98f5e8a25e6427b57ffe38ef2/packages/medusa/src/api/routes/store/products/list-products.ts#L143
I also think we support this in the service layer as well. As for the validation, that's a good point. I guess I got a bit lucky because we don't validate this class anywhere I think 😅